lib/kissable/ab.rb

```diff
@@ -19,8 +19,10 @@ def initialize(test_name, groups=nil, ratios=nil)
       validate_ratios
     end
 
-    def group(cookies)
-      @cookies = cookies
+    # Assigns a test group based on the 'abid' cookie or login (email).
+    # Takes either a cookies object (Hash-like) or login (String).
+    def group(object)
+      (object.is_a? String) ? (@login = object) : (@cookies = object)
```

Comment:
I've thought about this a bit, but I don't know that you would ever want to override it. And if you did, you'd probably want that logic to run before asking for the group. So you might do checks where you're only split testing a portion of your users.

You'd determine that precedence before passing either to the groups function, I believe